lib/Sema/TypeCheckConcurrency.cpp

```diff
@@ -2123,6 +2129,94 @@ namespace {
       }
     }
 
+    bool refineRequiredIsolation(ActorIsolation refinedIsolation) {
+      if (requiredIsolationLoc.isInvalid())
+        return false;
```

Comment:
For my education, because I've been bitten by checking Loc validity for logic before... Why are we returning early if that happens here? I think this can happen across modules, but the isolation is expected to be always in the same module here so that's why? Wanted to understand that bit to avoid missing handling similar cases 🤔

Reply:
We'll never compute the required isolation of a default argument or initializer expression across modules because we don't serialize expressions; we compute the required isolation in the original module and we need to serialize the required isolation to be checked at call-sites in other modules, so we should always have a source location here. However, I have not implemented nor tested the cross-module case yet 🙂

The reason I'm returning early here is because I'm using the validity of this source location to turn on the "required isolation" computation, so there's nothing to do here if the location isn't valid.

Reply:
I see, so really detecting cross module here kindof... Thanks for explaining